### PR TITLE
Leaderboard Screen UI

### LINKED ISSUE #15: Leaderboards UI

### PR DESCRIPTION
**Pull request by nbogdand**
Resolves #15

### CODE REVIEW

File: app/src/main/java/ip/team13/petowner/ui/leaderboard/LeaderboardViewModel.kt [hunk 1]
```diff
@@ -1,6 +1,29 @@
...
-class LeaderboardViewModel : ViewModel() {
+class LeaderboardViewModel(
+    private val leaderboardRepository: LeaderboardRepository
+) : BaseViewModel() {
+
+    val leaderboardType = ObservableField(LeaderboardType.FREE)
+
+    val person1 = ObservableField<LeaderboardEntry>()
+    val person2 = ObservableField<LeaderboardEntry>()
+    val person3 = ObservableField<LeaderboardEntry>()
```
Comment (andreimesina) [inline app/src/main/java/ip/team13/petowner/ui/leaderboard/LeaderboardViewModel.kt]: Why don't we use the actual type for each of these 4 fields and add the `@get:Bindable` annotation? It feels extra having to call `get()` and `set()` everytime we use them

Example:
```
@get:Bindable
val person1: LeaderboardEntry?
    get() = items[0]
```

**Review by andreimesina — CHANGES_REQUESTED** [review]

File: app/src/main/java/ip/team13/petowner/ui/leaderboard/LeaderboardViewModel.kt (hunk 1 above)
Comment (nbogdand) [reply app/src/main/java/ip/team13/petowner/ui/leaderboard/LeaderboardViewModel.kt]: Sure, will change

File: app/src/main/java/ip/team13/petowner/ui/leaderboard/LeaderboardViewModel.kt [hunk 2]
```diff
@@ -1,29 +1,38 @@
...
-    val leaderboardType = ObservableField(LeaderboardType.FREE)
-
-    val person1 = ObservableField<LeaderboardEntry>()
-    val person2 = ObservableField<LeaderboardEntry>()
-    val person3 = ObservableField<LeaderboardEntry>()
+    @get:Bindable
+    var leaderboardType: LeaderboardType = LeaderboardType.FREE
+        set(value) {
+            field = value
+            notifyChange()
```
Comment (andreimesina) [inline app/src/main/java/ip/team13/petowner/ui/leaderboard/LeaderboardViewModel.kt:18]: You don't need the `notifyChange`. All the `@Bindable` fields automatically notify the UI on any update

**Review by andreimesina — CHANGES_REQUESTED** [review]

File: app/src/main/java/ip/team13/petowner/ui/leaderboard/LeaderboardViewModel.kt (hunk 2 above)
Comment (nbogdand) [reply app/src/main/java/ip/team13/petowner/ui/leaderboard/LeaderboardViewModel.kt:18]: Does it notify all the fields? When the leaderboard type is changed, also items and person{1,2,3} will be changed.

Comment (andreimesina) [reply app/src/main/java/ip/team13/petowner/ui/leaderboard/LeaderboardViewModel.kt:18]: Ah ok. No it doesn't

**Review by andreimesina — DISMISSED** [review]

**Review by andreimesina — APPROVED** [review]

**Pull request merged**